frontend/app/controllers/Giraffe.scala

```diff
@@ -54,10 +71,13 @@ object Giraffe extends Controller {
       navigation = Seq.empty,
       customSignInUrl = Some((Config.idWebAppUrl / "signin") ? ("skipConfirmation" -> "true"))
     )
-    Ok(views.html.giraffe.contribute(pageInfo,maxAmount,countryGroup,isUAT, chosenVariants, cmp, intCmp, CreditCardExpiryYears(java.time.LocalDate.now().getYear, 10)))
+    val idUser = getIdUser(request).fold(Option.empty[IdentityUser])(Await.result(_, Duration("Inf")).some)
```

Comment:
Could this return a Future instead of Await infinite?  We should recover ID error rather than waiting forever.